tests/ui/lint/dead-code/false-positive-builder-with-self-128272.rs

```diff
@@ -0,0 +1,25 @@
+// Regression test for #128272
+// Tests that we do not consider a struct
+// dead code if it is constructed in a
+// const function taking `&self`
+
+
+//@ check-pass
+
+#![deny(dead_code)]
+
+pub struct Foo {
+    _f: i32
+}
+
+impl Foo {
+    // This function constructs Foo but in #128272
+    // the compiler warned that Foo is never constructed
+    pub const fn new(&self) -> Foo {
+        Foo { _f: 5 }
+    }
+}
+
+fn main() {
+    static _X: Foo = _X.new();
```

Comment:
I am not sure this is related. It's a private static item, and we won't visit it. But it can be treated as an example of constructing a static item using a method with the receiver.

Reply:
Yeah, I'm using `_X` only to reproduce the issue. This code is unlikely to occur in the real world.